packages/netstring/writer.js

```diff
@@ -1,51 +1,54 @@
 // @ts-check
 /// <reference types="ses"/>
 
-const COLON = ':'.charCodeAt(0);
-const COMMA = ','.charCodeAt(0);
+const COMMA_BUFFER = new Uint8Array([','.charCodeAt(0)]);
 
-// The initial buffer length should be small enough to not be an imposition in
-// the common case but large enough to avoid reallocation in the common case.
-// It also must be long enough to fit an ASCII decimal representation of the
-// length of the longest possible message.
-// This is a guess:
-const initialBufferLength = 128;
-
-const encoder = new TextEncoder();
+/** @param {number} length */
+const getLengthPrefixCharCodes = length =>
+  // eslint-disable-next-line no-bitwise
+  [...`${length | 0}:`].map(char => char.charCodeAt(0));
```

Comment:
Switching from `map` to a `for` loop may yield a measurable improvement.
```suggestion
const getLengthPrefixCharCodes = length => {
  // eslint-disable-next-line no-bitwise
  const text = String(length | 0);
  const codes = new Uint8Array(text.length + 1);
  let i;
  for (i = 0; i < codes.length; i += 1) {
    codes[i] = text.charCodeAt(i);
  }
  codes[i] = COLON;
}
```
Can also avoid the intermediate Array and go straight to Uint8Array.

Reply:
Given the size of the iteration I don't think the performance impact is measurable. Plus v8 can optimize these quite a bit these days.

I also wanted to avoid allocating an `ArrayBuffer` in the case where I just need to set the charCodes in an existing buffer. Not sure what's better, a temporary `ArrayBuffer` + `UInt8Array` or a temporary `Array`